presto-raptor/src/test/java/com/facebook/presto/raptor/storage/TestShardCompactor.java

```diff
@@ -116,67 +122,96 @@ public void testShardCompactorSorted()
         ShardCompactor compactor = new ShardCompactor(storageManager, READER_ATTRIBUTES);
 
         long transactionId = 1;
-        List<Long> columnIds = ImmutableList.of(3L, 7L, 2L, 1L, 5L);
-        List<Type> columnTypes = ImmutableList.of(BIGINT, VARCHAR, DOUBLE, DATE, TIMESTAMP);
 
+        List<Type> columnTypes = ImmutableList.of(BIGINT, VARCHAR, DATE, TIMESTAMP, DOUBLE);
+        List<Long> columnIds = ImmutableList.of(3L, 7L, 2L, 1L, 5L);
         List<Long> sortColumnIds = ImmutableList.of(1L, 2L, 3L, 5L, 7L);
         List<SortOrder> sortOrders = nCopies(sortColumnIds.size(), ASC_NULLS_FIRST);
         List<Integer> sortIndexes = sortColumnIds.stream()
                 .map(columnIds::indexOf)
                 .collect(toList());
 
-        // MAX_SHARD_ROWS = 100, so 10 shards of 10 rows each should be compacted to 1 shard
-        List<ShardInfo> inputShards = createShardsSorted(storageManager, columnIds, columnTypes, sortIndexes, sortOrders, 10, 10);
-        Set<UUID> inputUuids = ImmutableSet.copyOf(inputShards.stream()
-                .map(ShardInfo::getShardUuid)
-                .collect(toList()));
-        assertEquals(inputShards.size(), 10);
+        List<ShardInfo> inputShards = createSortedShards(storageManager, columnIds, columnTypes, sortIndexes, sortOrders, 2);
+        assertEquals(inputShards.size(), 2);
+
+        long totalRows = inputShards.stream().mapToLong(ShardInfo::getRowCount).sum();
+        long expectedOutputShards = computeExpectedOutputShards(totalRows);
 
+        Set<UUID> inputUuids = inputShards.stream().map(ShardInfo::getShardUuid).collect(toSet());
         List<ShardInfo> outputShards = compactor.compactSorted(transactionId, inputUuids, getColumnInfo(columnIds, columnTypes), sortColumnIds, sortOrders);
-        Set<UUID> outputUuids = ImmutableSet.copyOf(outputShards.stream()
+        List<UUID> outputUuids = outputShards.stream()
                 .map(ShardInfo::getShardUuid)
-                .collect(toList()));
-        assertEquals(outputShards.size(), 1);
+                .collect(toList());
+        assertEquals(outputShards.size(), expectedOutputShards);
 
         assertShardEqualsSorted(inputUuids, outputUuids, columnIds, columnTypes, sortIndexes, sortOrders);
     }
 
+    private static long computeExpectedOutputShards(long totalRows)
+    {
+        return ((totalRows % MAX_SHARD_ROWS) != 0) ? ((totalRows / MAX_SHARD_ROWS) + 1) : (totalRows / MAX_SHARD_ROWS);
```

Comment:
I think there is a trick to this... something like `(totalRows + MAX_SHARD_ROWS) / MAX_SHARD_ROWS`